proto/plugin.proto

```diff
@@ -739,21 +739,24 @@ message Deploy {
 
 // A platform resource that an operation (release/deployment) has created, depends on, or manages.
 message DeclaredResource {
+  // The id of the declared resource. Automatically generated
+  string id = 1;
+
   // platform-specific name of the resource type. i.e. instance, pod, auto-scaling group, etc
-  string type = 1;
+  string name = 2;
```

Comment:
Resource manager already has a "name" for each resource, and it's being used for exactly this purpose, so i'm renaming this new field to match it.

Reply:
From what I've read renaming isn't that bad if the number stays the same, but we're changing both here. Granted again the risk is probably low